model/histogram/float_histogram.go

```diff
@@ -192,6 +192,25 @@ func (h *FloatHistogram) Scale(factor float64) *FloatHistogram {
 //
 // This method returns a pointer to the receiving histogram for convenience.
 func (h *FloatHistogram) Add(other *FloatHistogram) *FloatHistogram {
+	if other.CounterResetHint != h.CounterResetHint {
+		// The outcome of adding an increment to a guage histogram will always be a GaugeType
+		if other.CounterResetHint == GaugeType && h.CounterResetHint != GaugeType {
```

Comment:
`&& h.CounterResetHint != GaugeType` is redundant because we already asserted that `other.CounterResetHint != h.CounterResetHint` above.